### ISSUE TITLE
Duplicate requires added when function present on requires 

### ISSUE THREAD

**Issue by billreed63** [maqetta/maqetta#1608]
A duplicate requires is added when a requires has a function example

```
<script type="text/javascript">
require(["dijit/dijit"]);
require(["dojo/parser"]);
require(["dojo/data/ItemFileReadStore"]);
require(["dojox/io/xhrScriptPlugin"],function(xhrScriptPlugin){xhrScriptPlugin("http://maqetta.org/sampleService","callback");});
require(["dojox/grid/DataGrid"]);
require(["dojox/io/xhrScriptPlugin"]);</script>
```

The root of the issue is in `davinci.ve.context.addHeaderScriptText(text)` what is passed into this method is "require(["dojox/io/xhrScriptPlugin"]);'  the code then checks to see if that text is already present at about line 2556

```
               var oldText = scriptText.getText();
        if (oldText.indexOf(text) === -1) {
            var newText = oldText + '\n' + text;
            scriptText.setText(oldText + '\n' + text);
            // XXX For some reason, <script> text is handled differently in the
            //   Model than that of other elements.  I think I only need to call
            //   setScript(), but the correct process should be to just update
            //   HTMLText. See issue #1350.
            scriptText.parent.setScript(oldText + '\n' + text);
        }
```

In this case the oldText is:

```
require(["dijit/dijit"]);
require(["dojo/parser"]);
require(["dojo/data/ItemFileReadStore"]);
require(["dojox/io/xhrScriptPlugin"],function(xhrScriptPlugin){xhrScriptPlugin("http://maqetta.org/sampleService","callback");});
require(["dojox/grid/DataGrid"]);"

```

So the test `if (oldText.indexOf(text) === -1) {` fails and the requires is added a second time...

To duplicate this issue:

1) Create html file
2) DnD a DataGrid
3) SmartInput choice Data from URL http://maqetta.org/sampleService 
4) Look at source and you should see the `require(["dojox/io/xhrScriptPlugin"], function...);`
5) DnD a another DataGrid, click ok for smart input
6) Look at the source and see the added `require(["dojox/io/xhrScriptPlugin"]);`

**Comment by billreed63** [comment]
I was also able to reproduce this with mobile..

1) Create a new HTML mobile page
2) DnD a header
3) Switch theme to custom for all devices
4) DnD anther header

Look at the source an notice a 2nd require for mobile has been added.

**Comment by peller** [comment]
see #1750

**Comment by jhpedemonte** [comment]
-> M6

**Comment by jhpedemonte** [comment]
Closing.  First issue was handled by #2158; 2nd fixed by #1750.
